### ISSUE TITLE
Question: Tooltip, Corss, Indicator Label does not shown if Tooltip component is not turned off/on in inspector.

### ISSUE THREAD

**Issue by HajiyevEl** [XCharts-Team/XCharts#315]
I think something is wrong with 'AxisHandler.UpdatePointerValue()' or pointer value does not get updated after i manually change axis split number, depending on xDataZoom current data quantity.
<details><summary>Screenshot</summary>
<p>

![Screenshot 2024-04-11 195915](https://github.com/XCharts-Team/XCharts/assets/70325881/0928b590-7b1c-4931-87dd-f130a309e9db)

</p>
</details> 

```c#
private void UpdateXAxisSplitNumber()
{
    var serie = chart.GetSerie(0);
    chart.GetDataZoomOfSerie(serie, out DataZoom xDataZoom, out DataZoom yDataZoom);

    var xAxis = chart.GetChartComponent<XAxis>(serie.xAxisIndex);

    xAxis.splitNumber = serie.GetDataList(xDataZoom).Count;
}
``` 
Also, please help me understand behaviour of indicator labels on axises when tooltip is set to Corss mode.

Problem is, when tooltip in Corss mode, indicator labels are hidden, even thought they are enabled on axises.

<details><summary>Indicator Labels Hidden (Play-mode)</summary>
<p>

![Screenshot 2024-04-12 115031](https://github.com/XCharts-Team/XCharts/assets/70325881/a2130c2b-2d61-4e1f-9f44-74121252dbe4)
![Screenshot 2024-04-12 114930](https://github.com/XCharts-Team/XCharts/assets/70325881/8366f18c-a23d-4ef6-8bf2-a48238a92e1e)
![Screenshot 2024-04-12 120512](https://github.com/XCharts-Team/XCharts/assets/70325881/ba0dcdf7-99c0-433a-9efd-317f7e618a6c)

</p>
</details> 

But if i disable and than enable Tooltip (Play-mode), they are shown and cannot be disabled.

<details><summary>Indicator Shown (Play-mode)</summary>
<p>

![Screenshot 2024-04-12 120557](https://github.com/XCharts-Team/XCharts/assets/70325881/24767543-6c18-4c0e-ab95-661e7016f832)
![Screenshot 2024-04-12 120643](https://github.com/XCharts-Team/XCharts/assets/70325881/fea1c89f-54ba-444a-aeec-5a5e87cabf16)
![Screenshot 2024-04-12 120759](https://github.com/XCharts-Team/XCharts/assets/70325881/3484a958-620f-40a9-b32c-32660af7b047)

</p>
</details> 

Somehow, when indicator labels shown/or hidden - enabling/or disabling Indicator labels (Play-mode) on axises does not change their active state. Is it intended?
By the way, i downloaded Demo project, and seen there same behavior in candlestick charts. If you enable/disable tooltip component (Play-mode), indicator labels are shown and cannot be disabled.

@monitor1394

**Comment by HajiyevEl** [comment]
Well, i think indicator label with dynamic zoom problem solved. Changed some code in TooltipHandler.cs :
```c#
private void SetTooltipIndicatorLabel(Tooltip tooltip, Axis axis, ChartLabel label)
{
 ...
    if (axis.IsCategory())
    {
// Old
        //var index = (int)axis.context.pointerValue;
        //var category = axis.GetData(index);
        //label.SetText(axis.indicatorLabel.GetFormatterContent(index, category));
// Edited
        var index = (int)axis.context.pointerValue;
        var dataZoom = chart.GetDataZoomOfAxis(axis);
        var category = axis.GetData(index, dataZoom);
        label.SetText(axis.indicatorLabel.GetFormatterContent(index, category));
    }
...
}
``` 
But still, can't make Tooltip Indicator Labels show itself without turning on/off Tooltip component. Any ideas?

**Comment by HajiyevEl** [comment]
Also, it seems that Indicator labels and labels of axis cannot be disabled by just un-ticking active component, you must change at least some value inside component (for example sprite color) from default to custom, otherwise enabling/disabling component will not do anything. Tested on 'background' settings of indicatorLabel of axis.

**Comment by HajiyevEl** [comment]
Hi again! Same problem: can't make Tooltip Indicator Labels show itself without turning off/on Tooltip component in inspector.
Please check this.

My Tooltip is set to Corss.
<details><summary>Tooltip Settings</summary>
<p>

![Screenshot 2024-04-27 130130](https://github.com/XCharts-Team/XCharts/assets/70325881/688ea626-34ca-49eb-aeeb-7134dacdc6a1)

</p>
</details> 

When tooltip component is initialized, Tooltip type becomes 'Line', and updates to Corss only later.
Also, 'm_LabelRoot' object is setting itself via 'ChartHelper.AddObject', but somewhere before code reaches 'UpdateTooltipIndicatorLabelText' - 'm_LabelRoot' again becomes null. 

<details><summary>Screenshots</summary>
<p>

![Screenshot 2024-04-27 125100](https://github.com/XCharts-Team/XCharts/assets/70325881/52bdddcd-d951-43e0-bae6-bb9566147e87)
![Screenshot 2024-04-27 125750](https://github.com/XCharts-Team/XCharts/assets/70325881/930a6c23-0e3f-4f2f-8990-3e0cf04a68fe)
![Screenshot 2024-04-27 130018](https://github.com/XCharts-Team/XCharts/assets/70325881/778c8f3d-2b11-4050-87af-5193d99ab0f3)

</p>
</details> 

Even tho it's spaghetti code, tooltip indicator labels is show without need for turning on/off Tooltip component in inspector. 
Can you please check in-depth about what causes this issue?


@monitor1394

**Comment by monitor1394** [comment]
> Hi again! Same problem: can't make Tooltip Indicator Labels show itself without turning off/on Tooltip component in inspector. Please check this.
> 
> My Tooltip is set to Corss.
> 
> Tooltip Settings
> ![Screenshot 2024-04-27 130130](https://private-user-images.githubusercontent.com/70325881/326167709-688ea626-34ca-49eb-aeeb-7134dacdc6a1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNDk2ODYsIm5iZiI6MTcxNDM0OTM4NiwicGF0aCI6Ii83MDMyNTg4MS8zMjYxNjc3MDktNjg4ZWE2MjYtMzRjYS00OWViLWFlZWItNzEzNGRhY2RjNmExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDAwMDk0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRkNzYyNTdlNjgwYTg1MjZlNTQ0MjE4MDRkNzM3ZTYwODc4OThhN2Q3NGUwOTVhZTkwYjYyODdlMjhkMzk2ZTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.onA42QOaqe4IVNCfkJk-tHGvISonyuFEQCFKQvRt3Js)
> 
> When tooltip component is initialized, Tooltip type becomes 'Line', and updates to Corss only later. Also, 'm_LabelRoot' object is setting itself via 'ChartHelper.AddObject', but somewhere before code reaches 'UpdateTooltipIndicatorLabelText' - 'm_LabelRoot' again becomes null.
> 
> Screenshots
> ![Screenshot 2024-04-27 125100](https://private-user-images.githubusercontent.com/70325881/326167702-52bdddcd-d951-43e0-bae6-bb9566147e87.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNDk2ODYsIm5iZiI6MTcxNDM0OTM4NiwicGF0aCI6Ii83MDMyNTg4MS8zMjYxNjc3MDItNTJiZGRkY2QtZDk1MS00M2UwLWJhZTYtYmI5NTY2MTQ3ZTg3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDAwMDk0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMwMzlhNzg2NmJhMTQyZmU3YjQ0YTFkMzEzOGNkMmU2MzA2MmYyNTdiMjlhOTVmYzZiMDcyNzBhZTZmMTJiMjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HLWBNo3qVfngBJ5UZHsnfctjhuNjq6-KghE0urM0o1c) ![Screenshot 2024-04-27 125750](https://private-user-images.githubusercontent.com/70325881/326167704-930a6c23-0e3f-4f2f-8990-3e0cf04a68fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNDk2ODYsIm5iZiI6MTcxNDM0OTM4NiwicGF0aCI6Ii83MDMyNTg4MS8zMjYxNjc3MDQtOTMwYTZjMjMtMGUzZi00ZjJmLTg5OTAtM2UwY2YwNGE2OGZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDAwMDk0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0MWRjNTg2MThjOTI2Y2Q1ZjQwODYyMGM4M2JjMDdhOGI2YTU3MzFkMzc5NDQ0OGEyZjM3ZWFhMzQ1MzMyZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZR1hD410U66OJhR4lMdae_ERdcdUR0XfFPSLCs0T6B4) ![Screenshot 2024-04-27 130018](https://private-user-images.githubusercontent.com/70325881/326167706-778c8f3d-2b11-4050-87af-5193d99ab0f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNDk2ODYsIm5iZiI6MTcxNDM0OTM4NiwicGF0aCI6Ii83MDMyNTg4MS8zMjYxNjc3MDYtNzc4YzhmM2QtMmIxMS00MDUwLTg3YWYtNTE5M2Q5OWFiMGYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI5VDAwMDk0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxMGZjNTUzYjA4ODlkYzFlMTJhNmYxNWI2OTQ2MjFmYWQ1ZWMwMGJmNDZjMGY5Y2Y2NzBhNGI4ZjhjN2Q2ZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fOzgUocn4b9VluL459ORpftdxIK11BrdLVzXgEItBYc)
> 
> I ended up adding label object manually. Even tho it's spaghetti code, tooltip indicator labels is shown without need for turning on/off Tooltip component in inspector. Can you please check in-depth about what causes this issue?
> 
> @monitor1394

I'm unable to replicate the issue you've described. Could you please provide detailed steps to reproduce it? 
Additionally, I attempted to set the type to 'Cross' both before and after execution, and in both cases, the operation performed normally.

**Comment by HajiyevEl** [comment]
> I'm unable to replicate the issue you've described. Could you please provide detailed steps to reproduce it? Additionally, I attempted to set the type to 'Cross' both before and after execution, and in both cases, the operation performed normally.

Steps to replicate:
1. Download Demo Project (https://github.com/XCharts-Team/XCharts-Demo)
2. Enter Play Mode
3. Select any chart with Corss type Tooltip (e.g. i selected Built-In Charts => Candlestick Chart => ShangHai Index) 
<details><summary>Select chart</summary>
<p>

![Screenshot 2024-04-29 111343](https://github.com/XCharts-Team/XCharts/assets/70325881/8abdd064-7128-437e-aec3-626a7591779a)

</p>
</details> 

4. Check if chart shows indicator labels. Check if all settings is correct. If no => go next
<details><summary>Check indicator labels and settings</summary>
<p>

![Screenshot 2024-04-29 111411](https://github.com/XCharts-Team/XCharts/assets/70325881/83796c20-a97a-4ade-a3ed-123f62ef2355)
![Screenshot 2024-04-29 111524](https://github.com/XCharts-Team/XCharts/assets/70325881/bcdfef0c-b89f-4e8d-ac79-a8395a3f00b4)
![Screenshot 2024-04-29 111549](https://github.com/XCharts-Team/XCharts/assets/70325881/7315a7b8-d406-47ee-9687-8178f608f046)
![Screenshot 2024-04-29 111606](https://github.com/XCharts-Team/XCharts/assets/70325881/1145ee1e-ee93-4694-a774-e3d3ff338a8a)

</p>
</details> 

5. Disable Tooltip component in inspector
<details><summary>Disable Tooltip</summary>
<p>

![Screenshot 2024-04-29 111629](https://github.com/XCharts-Team/XCharts/assets/70325881/ba240060-a52b-4774-b077-784f953c7699)

</p>
</details> 

6. Enable Tooltip component in inspector
<details><summary>Enable Tooltip</summary>
<p>

![Screenshot 2024-04-29 111653](https://github.com/XCharts-Team/XCharts/assets/70325881/b941af03-545b-4de4-bf53-ad51a33a2ec6)

</p>
</details> 

7. Check if indicator labels is shown
<details><summary>Check indicator labels</summary>
<p>

![Screenshot 2024-04-29 111717](https://github.com/XCharts-Team/XCharts/assets/70325881/923af0d2-0b39-416e-9d6f-61c5080a8088)

</p>
</details>

**Comment by monitor1394** [comment]
The issue has been fixed in version 3.11.0.
